gcloud/dns/zone.py

```diff
@@ -220,9 +228,11 @@ def _build_resource(self):
         """Generate a resource for ``create`` or ``update``."""
         resource = {
             'name': self.name,
-            'dnsName': self.dns_name,
         }
 
+        if self.dns_name is not None:
+            resource['dnsName'] = self.dns_name
```

Comment:
This is actually a bit of a weird case:  if the user doesn't pass `dns_name` or `description` to `ManagedZone.__init__`, and doesn't assign them, the API call in `ManagedZone.create` is going to fail anyway:  our usual rule is to let the back-end enforce those rules, so I didn't attempt any validation here:  I'm merely minimizing the resource sent with the `create` request (as the back-ends seem to prefer, at least based on how they return them).